zebrad/src/components/mempool/downloads.rs

```diff
@@ -282,12 +284,12 @@ where
 
             trace!(?txid, "transaction is not in best chain");
 
-            let next_height = match state.oneshot(zs::Request::Tip).await {
-                Ok(zs::Response::Tip(None)) => Ok(Height(0)),
+            let (tip_height, next_height) = match state.oneshot(zs::Request::Tip).await {
+                Ok(zs::Response::Tip(None)) => Ok((Height(0), Height(1))),
```

Comment:
Should `tip_height` be an `Option` here? i.e.
```rust
Ok(zs::Response::Tip(None)) => Ok((None, Height(0))),
```

Reply:
Yes, the correct implementation has `tip_height: Option<Height>`, because:
- `None` is an empty state
- `Height(0)` is a state containing only the genesis block

I don't think `next_height == 1` happens in production, but I think it might be required for some tests. If it isn't required for any tests, it would also be ok to return an error for an empty state.

I don't think we can panic, because we have tests that activate the mempool on startup. But feel free to try it!

Reply:
Fixed in [`99fa95e`](https://github.com/ZcashFoundation/zebra/pull/6154/commits/99fa95ebfdd2f468b984e0d9e166af0f4b148fb6).